### DISCUSSION TITLE
[Task]: Is it possible for data without labels?

### DISCUSSION THREAD

**Discussion by korkard** [open-edge-platform/anomalib#1895] [Q&A] [answered]
### What is the motivation for this task?

If the training data has a myriad of normal data and a small amount of abnormal data, and very small labelling test data

### Describe the solution you'd like

idk..

**Comment by blaz-r** [comment]
Hello. You can still train without the labels, but you won't be able to test the model.

**Comment by korkard** [comment]
Thanks for your reply.
Are you saying that I can train in a state where normal and abnormal are mixed?
I think good, bad is labels.

**Comment by blaz-r** [comment]
Sorry I understood the question incorrectly. All models implemented in Anomalib only utilize normal data for training, so you should be able to train the model using your good data, which does not require labels. The abnormal data can be used as part of the test set and shouldn't be mixed with good data.

**Comment by aradwan777** [comment]
hi to clarify, all models that exist in anomalib are unsupervised? Or in this case semi-supervised, I am not sure, but the learning is only done on the good images, that is the only data required, correct? Thank you, as I was confused if these were supervised(requiring masks for training, thus labeling for custom data) or not

**Comment by blaz-r** [comment, marked as answer]
Hello, that is correct. All models in currently Anomalib are unsupervised, using only good data for training. Labeled anomalous images are only used in test phase, where you require ground truth annotation to tell how good the model is performing.

**Comment by Abhijeet241093** [comment]
Is it possible, to test operation without using ground truth ?

**Comment by Sj-Yuan** [comment]
If you just need the result of anomaly detection but not localizaition, the ground truth is not required, you can just prepare some black image as the fake ground truth for test phase.

**Comment by samet-akcay** [comment]
@Abhijeet241093, you could refer to this section in the documentation
https://anomalib.readthedocs.io/en/latest/markdown/guides/how_to/data/custom_data.html#with-validation-and-testing-via-synthetic-anomalies

**Comment by ghost** [comment]
Hi, may I know what would happen if my testset contains both abnormal and normal images?

**Reply by blaz-r** [reply]
Hi, this is considered as a regular scenario in unsupervised learning, so everything will work as expected. Train set is always just normal images, while test set usually contains both anomalous and normal images.
